pkg/sliceutil/sliceutil.go

```diff
@@ -0,0 +1,41 @@
+// Package sliceutil provides a collection of slice functions.
+package sliceutil
+
+// Filter returns a new slice containing
+// only the elements in the input slice s for which the specified function f is true.
+func Filter[S ~[]E, E any](s S, f func(int, E) bool) S {
+	r := make(S, 0)
+
+	for k, v := range s {
+		if f(k, v) {
+			r = append(r, v)
+		}
+	}
+
+	return r
+}
+
+// Map returns a new slice that contains
+// each of the elements of the input slice s mutated by the specified function.
+func Map[S ~[]E, E any, U any](s S, f func(int, E) U) []U {
+	r := make([]U, len(s))
+
+	for k, v := range s {
+		r[k] = f(k, v)
+	}
```

Comment:
Just for readability, I would use `i`
```suggestion
	for i, v := range s {
		r[i] = f(k, v)
	}
```

Reply:
same for the other functions of the file

Reply:
It is using the same convention of the standard Google maps package:
https://cs.opensource.google/go/go/+/refs/tags/go1.21.0:src/maps/maps.go